src/stream.ts

```diff
@@ -3108,27 +3108,53 @@ export class Part extends Stream {
         return tempM.estimateStaffLength();
     }
 
-    systemWidthsAndBreaks(): [number[], number[]] {
-        const measureWidths = this.getMeasureWidths();
+    /**
+     * Calculate system breaks and update measure widths as necessary on
+     * account of the reiteration of clefs and key signatures on subsequent systems.
+     */
+    systemWidthsAndBreaks(setMeasureWidths: boolean = true): [number[], number[]] {
```

Comment:
prefer to pass a parameters object since otherwise we always need to document what "true" means.

Reply:
Perhaps better is to just break this method into two (or three). It's updating measure widths, calculating system widths, and calculating system breaks.

Reply:
No, I gave that an attempt and couldn't end up reducing any complexity.

Reply:
ah, too bad.  I hate it when good ideas go to waste.

Actually, I must have tried that at some point also.  By the time music21j was coming around, I already detested  `function doXAndYAndSometimesZ` type methods.